### DISCUSSION TITLE
WSS support?

### DISCUSSION THREAD

**Discussion by sloflo** [Textalk/websocket-php#175] [Q&A] [answered]
Will it work through TLS out of the box? I don't see any document soon regarding certificates and keys?

**Comment by sirn-se** [comment, marked as answer]
Yes, when specified protocol is `wss` it will use `ssl` for connection. Out of the box you would get standard php behavior, but it is also possible to pass a context to the constructor to set various SSL/TLS-related options.

* Example of using stream context in section [Constructor options](https://phrity.sirn.se/websocket/1.6.4/client).
* Available [SSL options](https://www.php.net/manual/en/context.ssl.php) in php.
* This repo is now maintained as [phrity/websocket](https://github.com/sirn-se/websocket-php)

**Reply by sloflo** [reply]
Appreciate your time!
